MoreLinq.Test/FromTest.cs

```diff
@@ -0,0 +1,125 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+
+namespace MoreLinq.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using NUnit.Framework;
+
+    class FromTest
+    {
+        [Test]
+        public void TestFromIsLazy()
+        {
+            var breakingFunc = BreakingFunc.Of<int>();
+            MoreEnumerable.From(breakingFunc);
+            MoreEnumerable.From(breakingFunc, breakingFunc);
+            MoreEnumerable.From(breakingFunc, breakingFunc, breakingFunc);
+            MoreEnumerable.From(breakingFunc, breakingFunc, breakingFunc, breakingFunc);
+        }
+
+        [TestCase(1)]
+        [TestCase(2)]
+        [TestCase(3)]
+        [TestCase(4)]
+        public void TestFromInvokesMethods(int numArgs)
+        {
+            var factories = new Func<int>[]
+            {
+                () => -2,
+                () => 4,
+                () => int.MaxValue,
+                () => int.MinValue,
+            };
+            IEnumerable<int> results;
+            int[] expectedResults;
+            switch (numArgs)
+            {
+                case 1:
+                    results = MoreEnumerable.From(factories[0]);
+                    expectedResults = new[] { factories[0]() };
+                    break;
+                case 2:
+                    results = MoreEnumerable.From(factories[0], factories[1]);
+                    expectedResults = new[] { factories[0](), factories[1]() };
+                    break;
+                case 3:
+                    results = MoreEnumerable.From(factories[0], factories[1], factories[2]);
+                    expectedResults = new[] { factories[0](), factories[1](), factories[2]() };
+                    break;
+                case 4:
+                    results = MoreEnumerable.From(factories[0], factories[1], factories[2], factories[3]);
+                    expectedResults = new[] { factories[0](), factories[1](), factories[2](), factories[3]() };
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(numArgs));
+            }
+
+            results.AssertSequenceEqual(expectedResults);
+        }
+
+        [TestCase(1)]
+        [TestCase(2)]
+        [TestCase(3)]
+        [TestCase(4)]
+        public void TestFromInvokesMethodsMultipleTimes(int numArgs)
+        {
+            var evals = new int[] { 0, 0, 0, 0 };
+            var factories = new Func<int>[]
+            {
+                () => { evals[0]++; return -2; },
+                () => { evals[1]++; return -2; },
+                () => { evals[2]++; return -2; },
+                () => { evals[3]++; return -2; },
+            };
+            IEnumerable<int> results;
+            switch (numArgs)
```

Comment:
This switch reads very long (nitpicking on readability) with lots of whitespace breaks when what we want to say is simply:

```c#
var results
    = numArgs == 1 ? MoreEnumerable.From(factories[0])
    : numArgs == 2 ? MoreEnumerable.From(factories[0], factories[1])
    : numArgs == 3 ? MoreEnumerable.From(factories[0], factories[1], factories[2])
    : numArgs == 4 ? MoreEnumerable.From(factories[0], factories[1], factories[2], factories[3])
    : throw new ArgumentOutOfRangeException(nameof(numArgs));
```

Even if you wish to maintain switch, I would suggest reformatting as follows:

```c#
switch (numArgs)
{
    case 1: results = MoreEnumerable.From(factories[0]); break;
    case 2: results = MoreEnumerable.From(factories[0], factories[1]); break;
    case 3: results = MoreEnumerable.From(factories[0], factories[1], factories[2]); break;
    case 4: results = MoreEnumerable.From(factories[0], factories[1], factories[2], factories[3]); break;
    default: throw new ArgumentOutOfRangeException(nameof(numArgs));
}
```